common/expected.hpp

```diff
@@ -46,6 +47,12 @@ class Expected {
 	};
 };
 
+template <typename ExpectedType, typename ErrorType>
+Expected<ExpectedType, ErrorType>::Expected() {
+	this->has_val_ = true;
+	new (&this->ex_) ExpectedType();
+}
```

Comment:
I'm not a big fan of this, it might be confusing for things like `ExpectedBool` or similar. I prefer explicit things over implicit. But then again, it's just up to us how (often) we use it.

Reply:
Good observation. I added this because I needed it in the test, but now that you pointed it out, I realized that it was passing because the mock object constructed a default return value behind my back, which is definitely *not* what we want. It should have been caught instead, which it was, as soon as I removed this. :+1: